docs/scripting-ref.rst

```diff
@@ -2230,3 +2230,126 @@ Trigger mouse release event in web page.
+* characters are literally represented
+* words within brackets represent function keys, like ``<Return>``, ``<Left>``
+  or ``<Home>``. See `Qt docs`__ for a full list of function keys. ``<Foo>``
+  will try to match ``Qt::Key_Foo``.
+
+__ http://doc.qt.io/qt-5/qt.html#Key-enum
+
+Following table shows some examples of macros and what they would generate on
+an input:
+
+======================    ===============
+Macro                     Result
+======================    ===============
+``Hello World``           ``HelloWorld``
+``Hello <Space> World``   ``Hello World``
+``< S p a c e >``         ``<Space>``
+``Hello <Home> DEL``      ``ello``
+``Hello <Backspace>``     ``Hell``
+======================    ===============
+
+Key events are not propagated immediately until event loop regains control,
+thus :ref:`splash-wait` must be called to reflect the events.
+
+.. _Qt key-enum: http://doc.qt.io/qt-5/qt.html#Key-enum
+
+.. _splash-send-text:
+
+splash:send_text
+----------------
+
+Send text as keyboard events to page context.
+
+**Signature:** ``splash:send_text(text)``
+
+**Parameters:**
+
+* text - string to be sent as keyboard events.
+
+**Returns:** nil
+
+**Async:**: no.
+
+Key events are not propagated immediately until event loop regains control,
+thus :ref:`splash-wait` must be called to reflect the events.
+
+This function is to be used to directly send text input as keyboard events.
+Whilst the macro syntax on :ref:`splash-send-keys` is useful for sending
+function keys, on some cases direct text input is more appropriate. It
+completely bypasses macro definition and sends text as character keys.
+
+This function in conjuction with :ref:`splash-send-keys` and covers most needs
+on keyboard input, such as filling in forms and submitting them.
+
+Examples:
+
+Example 1: focus first input, fill in a form and submit
+
+.. code-block:: lua
+
+    function main(splash)
+        assert(splash:go(splash.args.url))
+        assert(splash:wait(0.5))
+        splash:send_keys("<Tab>")
+        splash:send_text("zerocool")
+        splash:send_keys("<Tab>")
+        splash:send_text("hunter2")
+        splash:send_keys("<Return>")
+        -- note how this could be translated to
+        -- splash:send_keys("<Tab> zerocool <Tab> hunter2 <Return>")
+        assert(splash:wait(0))
+        -- ...
+    end
+
+We can't always assume that a TAB will focus the input we want or an Enter will
+submit a form. By using :ref:`splash-mouse-click` we can focus an input. The
+following example will click an input, fill in a form and submit it. It assumes
+there are two arguments passed to splash, `username` and `password`.
+
+.. code-block:: lua
+
+    function main(splash)
+        local get_elem_pos = splash:jsfunc([[
+            function (selector) {
+                var elem = document.querySelector(selector);
+                var rect = elem.getBoundingClientRect();
+                return {"x": rect.left, "y": rect.top}
+            }
```

Comment:
Just found out another thing about using focus (not really important).

Since focus happens on JS and not on the event loop, a wait must be issued after each send keys, like

```
-- This will input barbaz on assumed input bar
focus('foo')
send_text('bar')
focus('bar')
send_text('baz')
assert(splash:wait(0))
```

```
-- This does what we want
focus('foo')
send_text('bar')
assert(splash:wait(0))
focus('bar')
send_text('baz')
assert(splash:wait(0))
```

Anyway, I have updated the example to include both a focus solution and a click solution, at the same time (focus for selecting an input and click for pressing the send button).